docs/docs/clients.md

```diff
@@ -0,0 +1,246 @@
+
+```go
+package main
+
+import (
+    "context"
+    "encoding/hex"
+    "fmt"
+    "github.com/drand/drand/client"
+    "github.com/drand/drand/client/http"
+)
+
+const url = "https://drand.cloudflare.com"
+
+var chainHash, _ = hex.DecodeString("TODO-LOE-CHAIN-HASH")
+
+func main() {
+    c, err := client.New(
+        client.From(http.ForURLs([]string{url}, chainHash)...),
+        client.WithChainHash(chainHash),
+    )
+}
+```
+
+The `ForURLs` helper creates multiple HTTP clients from a list of URLs. Alternatively you can use the `New` or `NewWithInfo` constructor to create clients.
+
+:::tip
+Provide multiple URLs to enable failover and speed optimised URL selection.
+:::
+
+Consult the [HTTP client API reference](https://pkg.go.dev/github.com/drand/drand/client/http) for more information.
+
+#### gRPC
+
+The [gRPC](https://grpc.io/) client connects to a drand gRPC endpoint to fetch randomness. The gRPC client has some advantages over the HTTP client - it is more compact on-the-wire and supports streaming and authentication.
+
+```go
+package main
+
+import (
+    "context"
+    "encoding/hex"
+    "fmt"
+    "github.com/drand/drand/client"
+    "github.com/drand/drand/client/grpc"
+)
+
+const (
+    grpcAddr  = "example.drand.grpc.server:4444"
+    certPath  = "/path/to/drand-grpc.cert"
+)
+
+var chainHash, _ = hex.DecodeString("TODO-LOE-CHAIN-HASH")
+
+func main() {
+    gc, err := grpc.New(grpcAddr, certPath, false)
+
+    c, err := client.New(
+        client.From(gc),
+        client.WithChainHash(chainHash),
+    )
+}
+```
+
+A path to a file that holds TLS credentials for the drand server is required to validate server connections. Alternatively set the final parameter to `true` to enable _insecure_ connections (not recommended).
+
+Consult the [gRPC client API reference](https://pkg.go.dev/github.com/drand/drand/client/grpc) for more information.
+
+#### Libp2p PubSub
+
+The [libp2p PubSub](https://docs.libp2p.io/concepts/publish-subscribe/) client is a special type of client called a "watcher" that receives randomness rounds by subscribing to a libp2p PubSub topic.
+
+:::warning
+This client can only be used to `Watch` for new randomness rounds and `Get` randomness rounds it has previously seen that are still in the cache.
+
+If you need to `Get` arbitrary rounds from the chain then you must combine this client with a [HTTP](#http) or [gRPC](#grpc) client.
+:::
+
+Drand _does not_ publish randomness rounds over libp2p PubSub by default but provides a [relay tool](/operate/drand-cli/#drand-relay-gossip) that performs this task. The [League of Entropy](https://blog.cloudflare.com/league-of-entropy/) provide the following public drand libp2p PubSub relays:
+
+- `/dns4/todo/p2p/QmPeerID`
```

Comment:
@hsanjuan will we run public libp2p pubsub relays for the LoE? If so what's the dnsaddr likely to be? `/dnsaddr/drand.protocol.ai`?